plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeMetaModel.java

```diff
@@ -137,4 +137,65 @@ public DBCQueryTransformer createQueryTransformer(@NotNull DBCQueryTransformType
         }
         return null;
     }
+
+    @Override
+    public JDBCStatement prepareUniqueConstraintsLoadStatement(
+        @NotNull JDBCSession session,
+        @NotNull GenericStructContainer owner,
+        @Nullable GenericTableBase forParent
+    ) throws SQLException {
+        boolean recognizeWildCards = isRecognizeWildCards(session, owner);
+        GenericSchema schema = owner.getSchema();
+        String schemaName = getSchemaNameForPattern(session, recognizeWildCards, schema);
+        String tableName = getTableNameForPattern(session, owner, forParent, recognizeWildCards);
+        return session.getMetaData()
+            .getPrimaryKeys(owner.getCatalog() == null ? null : owner.getCatalog().getName(), schemaName, tableName)
+            .getSourceStatement();
+    }
+
+    @Override
+    public JDBCStatement prepareForeignKeysLoadStatement(
+        @NotNull JDBCSession session,
+        @NotNull GenericStructContainer owner,
+        @Nullable GenericTableBase forParent
+    ) throws SQLException {
+        boolean recognizeWildCards = isRecognizeWildCards(session, owner);
+        GenericSchema schema = owner.getSchema();
+        String schemaName = getSchemaNameForPattern(session, recognizeWildCards, schema);
+        String tableName = getTableNameForPattern(session, owner, forParent, recognizeWildCards);
+        return session.getMetaData()
+            .getImportedKeys(owner.getCatalog() == null ? null : owner.getCatalog().getName(), schemaName, tableName)
+            .getSourceStatement();
+    }
+
+    private boolean isRecognizeWildCards(@NotNull JDBCSession session, @NotNull GenericStructContainer owner) throws SQLException {
```

Comment:
Maybe just `supportsWildcards`?

Reply:
I do not see any reason for this, but ok